extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/SessionCachingMetadataDomainRepository.java

```diff
@@ -515,4 +582,79 @@ public void addLocalizationFile( String domainId, String locale, InputStream inp
           .storeAnnotationsXml( domainId, annotationsXml );
     }
   }
+
+  /**
+   * Asynchronously populate the cache for the specified domain identifiers.
+   * @param domainIds collection of domain identifiers.
+   * @param pentahoSession session used to set in spawned threads
+   * @return asynchronously running thread
+   */
+  Thread asyncPopulateCacheDomain( Collection<String> domainIds, IPentahoSession pentahoSession ) {
```

Comment:
[#7 professionalize code for asynchronous populate cache](https://jira.pentaho.com/browse/BACKLOG-36132)